tsconfig.json

```diff
@@ -0,0 +1,70 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */
+    "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
```

Comment:
Should we include some default libs here, or manage this in our boilerplates? e.g.:
```js
"es2017",
"dom"
```

Reply:
Probably makes sense to have some defaults here. I can imagine not using "dom" when writing libraries, but does having it anyway have any downside?

Reply:
So we have 2 options:
- leave this default, and require projects/boilerplates to set this explicitly (opt-in)
- include `esnext`, `dom` to be future-compatible for most of our use cases, and allow projects to change or opt-out of this setting (opt-out)

Including more by default can only do harm of a project doesn't include all the polyfills for the given version, and developer uses it without testing their code.
But this could also happen if we include `es2019` if we need some of the features, but haven't included polyfills for all of them.

Using `esnext` here means we don't have to update it every year, and kan keep using the newest features combined with babel-preset-env in our projects.

Reply:
I like the 'opt-out' option, but I would suggest the following list: 
```
      "ESNext",
      "DOM",
      "DOM.Iterable",
      "ScriptHost",
      "ESNext.AsyncIterable"
```